### PR TITLE
backup: fix background panic in `endpoint::tests::test_adjust_thread_pool_size`

### LINKED ISSUE #11548: bkwkr paniced in test_adjust_thread_pool_size
## Bug Report

run `make dev` and found
```
thread 'bkwkr' panicked at 'Cannot drop a runtime in a context where blocking is not allowed. This happens when a runtime is dropped from within an asynchronous context.', /data4/juncen/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.12.0/src/runtime/blocking/shutdown.rs:51:21
```
### What version of TiKV are you using?
master

### Steps to reproduce

run `make dev` on master

### What did you expect?
do not panic bkwkr
### What did happened?
bkwkr paniced though test passed🤔

### PR DESCRIPTION
**Pull request by YuJuncen**
Signed-off-by: yujuncen <yujuncen@pingcap.com>

### What is changed and how it works?

Close #11548

What's Changed:

See comments in #11548

### Check List 

Tests 

- Manual test (add detailed scripts or steps below)
image

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by NingLin-P — APPROVED** [review]

**Review by 3pointer — APPROVED** [review]
LGTM

File: components/backup/src/endpoint.rs [hunk 1]
```diff
@@ -1603,5 +1600,18 @@ pub mod tests {
...
+        assert_eq!(responses.len(), 3);
+
+        // for testing whether dropping the pool before all tasks finished causes panic.
+        // but the panic must be checked manually... (It may panic at tokio runtime threads...)
+        let mut pool = ControlThreadPool::new();
+        pool.adjust_with(1);
+        pool.spawn(async { tokio::time::sleep(Duration::from_millis(100)).await });
+        pool.adjust_with(2);
+        drop(pool);
+        std::thread::sleep(Duration::from_millis(150));
```
Comment (overvenus) [inline components/backup/src/endpoint.rs:1615]: Why sleep 150ms?

Comment (YuJuncen) [reply components/backup/src/endpoint.rs:1615]: Just for trigger the panic(or the test exist so fast and the panic cannot happen...)

**Pull request merged**